sdk/timeseriesinsights/Azure.Iot.TimeSeriesInsights/src/TimeSeriesInsightsClient.cs

```diff
@@ -239,5 +246,813 @@ public virtual Response<TimeSeriesModelSettings> UpdateModelSettingsDefaultTypeI
+            CancellationToken cancellationToken = default)
+        {
+            using DiagnosticScope scope = _clientDiagnostics
+                .CreateScope($"{nameof(TimeSeriesInsightsClient)}.{nameof(CreateOrReplaceTimeSeriesInstances)}");
+            scope.Start();
+
+            try
+            {
+                Argument.AssertNotNullOrEmpty(timeSeriesInstances, nameof(timeSeriesInstances));
+
+                InstancesBatchRequest batchRequest = new InstancesBatchRequest();
+
+                foreach (TimeSeriesInstance instance in timeSeriesInstances)
+                {
+                    batchRequest.Put.Add(instance);
+                }
+
+                Response<InstancesBatchResponse> executeBatchResponse = await _timeSeriesInstancesRestClient
+                    .ExecuteBatchAsync(batchRequest, _clientSessionId, cancellationToken)
+                    .ConfigureAwait(false);
+
+                // Extract the errors array from the response. If there was an error with creating or replacing one of the instances,
+                // it will be placed at the same index location that corresponds to its place in the input array.
+                IEnumerable<InstancesOperationError> errorResults = executeBatchResponse.Value.Put.Select((result) => result.Error);
+
+                return Response.FromValue(errorResults.ToArray(), executeBatchResponse.GetRawResponse());
+            }
+            catch (Exception ex)
+            {
+                scope.Failed(ex);
+                throw;
+            }
+        }
+
+        /// <summary>
+        /// Creates Time Series instances synchronously. If a provided instance is already in use, then this will attempt to replace the existing
+        /// instance with the provided Time Series Instance.
+        /// </summary>
+        /// <param name="timeSeriesInstances">The Time Series instances to be created or replaced.</param>
+        /// <param name="cancellationToken">The cancellation token.</param>
+        /// <returns>
+        /// List of error objects corresponding by position to the <paramref name="timeSeriesInstances"/> array in the request.
+        /// A <seealso cref="InstancesOperationError"/> object will be set when operation is unsuccessful.
+        /// </returns>
+        /// <seealso cref="CreateOrReplaceTimeSeriesInstancesAsync(IEnumerable{TimeSeriesInstance}, CancellationToken)">
+        /// See the asynchronous version of this method for examples.
+        /// </seealso>
+        /// <exception cref="ArgumentNullException">
+        /// The exception is thrown when <paramref name="timeSeriesInstances"/> is <c>null</c>.
+        /// </exception>
+        /// <exception cref="ArgumentException">
+        /// The exception is thrown when <paramref name="timeSeriesInstances"/> is empty.
+        /// </exception>
+        public virtual Response<InstancesOperationError[]> CreateOrReplaceTimeSeriesInstances(
```

Comment:
Something I just noticed now, did you mean to create a separate API for replace or combined here?

Looking at code at line: 772